mongodb.js

```diff
@@ -0,0 +1,102 @@
+const { MongoClient, Timestamp } = require("mongodb");
+
+module.exports = { db_connect, db_setup, new_print_job, new_workflow, new_workflow_step, new_simulation_report };
+
+
+async function db_connect(){
+	const uri = "mongodb://localhost:27017/HP";
+	const client = new MongoClient(uri);
+	console.log("Connecting to MongoDB...");
+	await client.connect();
+	console.log("Database connected");
+	const database = client.db('HP');
+	return [client, database];
+}
+
+
+// DELETES all existing collections, recreates them, then inserts
+// some dummy data
+async function db_setup(database){
+	//Drop all collections
+	const old_collections = await database.listCollections().toArray();
+	Promise.all(old_collections.map((c) => database.collection(c.name).drop()));
+
+	//Create collections
+	const collections = ["PrintJob", "Workflow", "WorkflowStep", "SimulationReport"];
+	Promise.all(collections.map((c) => database.createCollection(c)));
+
+	//Insert dummy data
+	const pj_id = await new_print_job(database, "PrintJob 1", 5, ["RP 1"]);
+	const ws_id = await new_workflow_step(database, "WorkflowStep 1", null, null, 10, 7);
+	const wf_id = await new_workflow(database, "Workflow 1", [ws_id]);
+	await new_simulation_report(database, pj_id, wf_id, 3, 4);
+}
+
+
+async function insert(database, collection_name, doc){
+	const collection = database.collection(collection_name);
+	const insert = await collection.insertOne(doc);
+	if(insert.acknowledged){
+		//TODO: Do Stuff
+	} else { 
+		//TODO: Do other stuff
+	}
+	return insert.insertedId;
+}
+
+
+function check_null(args){
+	for(const arg of args){
+		if(arg == null) throw new Error("Invalid argument");
+	}
+}
+
+
+//title is a string, page_count is an int, and rasterization_profile is an array of strings
```

Comment:
Instead of this description, we should follow [JS documentation standards](https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/). Place TODO's inside the function inside of outside of it.

So here, it would be (* means any variable type):
```
/**
 * @param {*} database The MongoDB connection.
 * @param {string} title The title of the new print job.
 * @param {int} pageCount The page count of the new print job.
 * @param {*} rasterizationProfile The rasterization profile of the new print job.
 */
```